src/main/java/org/rumbledb/compiler/StaticContextVisitor.java

```diff
@@ -127,39 +127,58 @@ public StaticContext visitFunctionCall(FunctionCallExpression expression, Static
     // region FLWOR
     @Override
     public StaticContext visitFlowrExpression(FlworExpression expression, StaticContext argument) {
-        StaticContext result = this.visit(expression.getStartClause(), argument);
-        for (Clause clause : expression.getContentClauses()) {
+        Clause clause = expression.getReturnClause().getFirstClause();
+        StaticContext result = this.visit(clause, argument);
+        while (clause != null) {
             result = this.visit(clause, result);
+            clause = clause.getNextClause();
         }
-
-        result = this.visit(expression.getReturnClause(), result);
         return result;
```

Comment:
I have a feeling that returning 'argument' here instead of result would be more correct. I think return of this function means going out of scope for the FLWOR clause which means all the variables defined in there should go out of scope. Hence the 'argument' should be returned instead of 'result' to facilitate this.

Reply:
Very good catch too. Now it is easier to see. Let's see if tests still pass.

Reply:
I also did the same change for other visit calls doing the same mistake. It doesn't change anything anyway: results from visitABC() on expressions are always discarded, it is only used for clauses. But it is more correct.